explorer/src/pages/Mixnodes/index.tsx

```diff
@@ -46,24 +49,58 @@ const columns = [
 
 export const PageMixnodes: React.FC = () => {
   const { mixnodes } = useContext(MainContext);
-  const [rows, setRows] = React.useState();
+  const [filteredMixnodes, setFilteredMixnodes] = React.useState<MixNodeResponse | undefined>()
+  const [pageSize, setPageSize] = React.useState<string>("50");
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const st = event.target.value;
+    if (st === '' && mixnodes?.data) {
+      const formatted = mixnodeToGridRow(mixnodes?.data)
+      setFilteredMixnodes(formatted)
+    } else {
+      const filtered = mixnodes?.data && mixnodes?.data.filter((m) => {
+        if (
+          m.location?.country_name.toLowerCase().includes(st) ||
+          m.mix_node.identity_key.toLocaleLowerCase().includes(st) ||
+          m.owner.toLowerCase().includes(st)
+        ) {
+          return m;
+        }
+      })
+      if (filtered) {
+        const formatted = mixnodeToGridRow(filtered)
+        setFilteredMixnodes(formatted)
+      }
+    }
+  }
+
+  const handlePageSize = (event: SelectChangeEvent<string>) => {
+    setPageSize(event.target.value);
+  };
 
   React.useEffect(() => {
-    if (mixnodes?.data !== undefined) {
-      const formattedRows = mixnodeToGridRow(mixnodes.data);
-      setRows(formattedRows);
-    };
-  }, [mixnodes])
+    if (mixnodes?.data) {
+      const formatted = mixnodeToGridRow(mixnodes?.data)
+      setFilteredMixnodes(formatted)
+    }
+  }, [mixnodes]);
+
   return (
     <>
       <Typography sx={{ marginBottom: 1 }} variant="h5">
         Mixnodes
       </Typography>
+      <TableToolbar
+        onChangeSearch={handleSearch}
+        onChangePageSize={handlePageSize}
+        pageSize={pageSize}
+      />
       <UniversalDataGrid
         loading={mixnodes?.isLoading}
         columnsData={columns}
-        rows={rows}
+        rows={filteredMixnodes}
```

Comment:
had an idea! what if we make this
```
rows={mixnodeToGridRow(filteredMixnodes)}
```
Then you won't have to call this function to format the data every time you want to call `setFilteredMixnodes`
*filteredMixnodes can remain as the current type*